pkg/daemon/certs/certs.go

```diff
@@ -106,69 +182,102 @@ func splitAtPEMCert() func(data []byte, atEOF bool) (advance int, token []byte,
 	}
 }
 
-// ensureCertsAreImported imports the expected certificates into the instance's root system trust store, if not already
-// present. Returns a boolean if any certificates were imported
-func ensureCertsAreImported(expectedCerts []*x509.Certificate) (bool, error) {
-	// Open the root certificate store
-	systemStore, err := windows.CertOpenStore(windows.CERT_STORE_PROV_SYSTEM, 0, 0,
-		windows.CERT_SYSTEM_STORE_LOCAL_MACHINE, uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr("ROOT"))))
+// fileExists checks if the file at the given path exists. Returns true if it exists, false if not, and error otherwise.
+func fileExists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true, nil
+	} else if errors.Is(err, os.ErrNotExist) {
+		return false, nil
+	} else {
+		return false, fmt.Errorf("Failed to check if file %s exists: %w", path, err)
+	}
+}
+
+// updateImportedCABundle updates the previously imported CA bundle file, if needed.
+func updateImportedCABundle(caBundlePath string, certChange bool) error {
+	if caBundlePath == "" {
+		// When proxy is deconfigured, ensure the file is deleted to avoid leaking certs
+		return ensureFileIsRemoved(importedCABundleFile)
+	}
+	if certChange {
+		klog.Infof("copied file contents from %s to %s", caBundlePath, importedCABundleFile)
+		// When certs have been reconciled, update the previously imported CA trust bundle to hold the new cert bundle
+		return copyFile(caBundlePath, importedCABundleFile)
+	}
+	return nil
+}
+
+// ensureFileIsRemoved deletes the file at the given path. No-op if the file already does not exist
+func ensureFileIsRemoved(path string) error {
+	exists, err := fileExists(path)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return nil
+	}
+	klog.Infof("removing file %s", path)
```

Comment:
Nit: I am starting to see a pattern where we add logs saying we are doing X but no log saying it succeeded. This is not great when debugging or for the end user. Best is to either follow up with a message saying that the file has been removed or just log when we are actually removed the file after the fact. I had a tough time with this in the HNS removal area where we say we are removing the networks but without an indicator of success.